### PR TITLE
fix: allow any standard HTML attributes as props in the NanoClamp type

### LINKED ISSUE #26: TypeScript problem: Property className does not belong to the Props type definition
I'm using your library in a React-TypeScript project. When adding a `className` attribute to the `NanoClamp` tag, the TypeScript compiler complains about the unknown property `className`.

Code:
```jsx
<NanoClamp text={s.value} lines={2} className="custom-class" />
```

TypeScript error:
```
src/App.tsx:195:61 - error TS2322: Type '{ text: string; lines: number; className: string; }' is not assignable to type 'IntrinsicAttributes & Props'.
  Property 'className' does not exist on type 'IntrinsicAttributes & Props'.
```

### PR DESCRIPTION
**Pull request by obecker**
+ Fixes https://github.com/microlinkhq/nanoclamp/issues/26

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { createElement, useRef, useEffect, useCallback, useMemo } from 'react'
+import { createElement, useRef, useEffect, useCallback, useMemo, HTMLAttributes } from 'react'
 
-interface Props {
+interface Props extends HTMLAttributes<HTMLElement>{
```
Comment (breadadams) [inline src/index.ts]: **suggestion:** Can we save on an import and use the global namespace here?

```suggestion
interface Props extends React.HTMLAttributes<HTMLElement>{
```

File: src/index.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ const NanoClamp = ({
   lines = 3,
   text,
   ...props
-}: Props): JSX.Element | null => {
+}: Props): React.ReactElement | null => {
```
Comment (breadadams) [inline src/index.ts:39]: **thought:** I think we can do without this annotated return type and just infer it:

```suggestion
}: Props) => {
```

**Review by breadadams — COMMENTED** [review]
Thanks for the contribution @obecker, couple of minor suggestions

File: src/index.ts (hunk 1 above)
Comment (obecker) [reply src/index.ts]: Done

File: src/index.ts (hunk 2 above)
Comment (obecker) [reply src/index.ts:39]: Done

Comment (obecker) [reply src/index.ts:39]: I just saw in the other PR that your lint configuration requires all functions to have a return type.
https://github.com/microlinkhq/nanoclamp/actions/runs/8409288351/job/23027852174?pr=29
So I will add the return type again.

**Comment by Kikobeats** [comment]
@obecker Thanks for this! This library is mainly in maintenance mode, but even though we are not actively developing this library, any PR that improves the library (like this one) will be merged and released, so please feel free to open as many PRs as you like!

**Pull request merged**